sycl/include/sycl/vector_preview.hpp

```diff
@@ -755,12 +438,12 @@ template <typename Type, int NumElements> class vec {
   template <typename convertT,
             rounding_mode roundingMode = rounding_mode::automatic>
   std::enable_if_t<
-      std::is_same_v<vec_data_t<DataT>, vec_data_t<convertT>> ||
-          std::is_same_v<detail::ConvertToOpenCLType_t<vec_data_t<DataT>>,
-                         detail::ConvertToOpenCLType_t<vec_data_t<convertT>>>,
+      std::is_same_v<element_type_for_vector_t<DataT>, element_type_for_vector_t<convertT>> ||
+          std::is_same_v<detail::ConvertToOpenCLType_t<element_type_for_vector_t<DataT>>,
+                         detail::ConvertToOpenCLType_t<element_type_for_vector_t<convertT>>>,
```

Comment:
I think this code predates C++17. Can we just inline it using `if constexpr` now?

Reply:
Done!